tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,178 @@
+   the state machine automatically performs the aggregation. The subsequent
+   block proposer checks if the aggregation was performed. If so, it adds 
+   the result to the block it is proposing. In either case, it also adds the
+   PVSS transcripts to the block header. If aggregation was successful, these
+   act as verifications. Otherwise, the next block proposer will need the 
+   information to try and complete the  protocol in the next block. For any
+   PVSS transcript that fails verification, a `Complain` message is also 
+   included in the header.
+ - `Aggregate`: This simply needs to be verified and the resulting key added
+   to the DKG instance along with the input PVSS transcripts. If validation
+   of the aggregated key fails, a `Complain` should be issued.
+
+While every validator must send and react to `Deal` messages, `Aggregate` 
+must be made by the current block proposer if their current DKG state machine 
+is in the  appropriate state and/or the appropriate conditions are met. Thus 
+block proposers must check the state machines of their active DKG instances.
+
+## ABCI++
+
+The creation of the public key and the shared secrets requires ABCI++ as we
+need control over block preparation as well extra data from the Vote Extension
+phase. 
+
+We list the phase of ABCI++ in which each of the above actions takes place.
+
+ - `Deal`: `Deal` happens during the Vote Extension phase.
+ - `Aggregate`: This is done during the Prepare Proposal phase (it could be
+   done earlier if validators know they are likely to be the next proposer)
+   and will be  included in the header of the proposed block. The final result 
+   should be verified by validators. If validation fails, the next block 
+   proposer can try to aggregate the the PVSS transcripts correctly and 
+   include the result in the header of the subsequent block. Furthermore, 
+   `Complains` may be issued against the validator that incorrectly
+   aggregated the PVSS transcripts. Alternatively, the block could be
+   rejected by other validators and the next round begins immediately.
+   The faulty proposer thus loses out on the propose block rewards.
+
+## Complaints
+
+Complaints are encrypted transactions that validators broadcast when they
+observe misbehavior from other validators during the protocol. The current
+causes for complaint are as follows:
+ * `PVSS_Complaint`: During VerifyVoteExtension, the PVSS transcript from a validator does not
+   pass validation
+ * `Aggregate_Complaint`: The aggregated key in a proposed block is not valid
+ 
+These transactions must include a proofs of malfeasance. The necessary 
+proofs for the above complaints are as follows:
+ * `PVSS_Complaint`: The transcript that failed verification
+ * `Aggregate_Complaint`: The alleged incorrect key and the PVSS transcripts
+    that should have been used to derive it. 
+
+The above proofs can be used to verify that the protocol was not adhered to.
```

Comment:
these are just verified by the state machine, yes?

Reply:
Again, this is outdated as complaints have been reworked